openapi-style-guide.md

```diff
@@ -0,0 +1,208 @@
+
+### Request body
+
+The request body for a patch operation should be `application/merge-patch+json` content type.
+
+### Response body
+
+All success responses except 202 and 204 should define a response body.
+
+Responses for status codes 202 and 204 should have no response body.
+
+A delete operation should have a 204 response.
+
+For a path with a "create" operation (put or patch that returns 201),
+the 200 response of get, put, and patch, if present, should have the same response body schema
+as the create operation 201 response.
+
+### Support for pagination
+
+To support pagination:
+- The operation should have the `x-ms-pageable` annotation
+- The operation response should contain a top-level `value` property of type array and required
+- The operation response should contain a top-level `nextLink` property of type string and optional
+- If the operation has a `skip` parameter, it must be an integer and optional
+- If the operation has a `top` parameter, it must be an integer, optional, and have a documented default and maximum value
+- If the operation has a `maxpagesize` parameter, it must be an integer, optional, and have a documented default and maximum value
+
+### Error response
+
+All operations should have a "default" (error) response with a response body that conforms to the Azure API Guidelines.
+
+All `4xx` and `5xx` responses should specify `x-ms-error-response: true` except for `404` response of HEAD operation.
+
+Every error response should include the `x-ms-error-code` response header in the `headers` of the response.
+
+Example:
+```
+      "400": {
+        "description": "Bad Request",
+        "headers": {
+          "x-ms-error-code": {
+            "type": "string",
+            "description": "The error code."
+          }
+        },
+        "schema": {
+          "$ref": "#/definitions/ErrorResponse"
+        },
+        "x-ms-error-response": true
+      },
+```
+
+### Response headers
+
+A 202 response should include the `Operation-Location` response header in the `headers` of the response.
+
+Example:
+```
+    "responses": {
+      "202": {
+        "description": "The service has accepted the request and will start processing later.",
+        "headers": {
+          "Operation-Location": {
+            "description": "URL to query for status of the operation.",
+            "type": "string"
+          }
+        }
+      },
+```
+
+<!-- --------------------------------------------------------------- -->
+
+## Parameters
+
+### Parameter names
+
+All parameter names for an operation -- including parameters defined at the path level -- should be case-insensitive unique.
+
+### Descriptions
+
+Every parameter should have a description.
+
+### Format
+
+Format must be one of the values [defined by OpenAPI][openapi-data-types] and recognized by the Azure tooling.
+
+Integer properties must specify a format of either `int32` or `int64`.
+
+### Default
+
+Required parameters should not specify a default value.
```

Comment:
Except for api-version?

Reply:
I think for `api-version` we would want them to specify an [`x-ms-client-default`](https://github.com/Azure/autorest/tree/main/docs/extensions#x-ms-client-default), which would be a good thing to add into the API versioning section.

Reply:
Good point.